packages/core/test/util/runtime.test.ts

```diff
@@ -514,6 +531,96 @@ test('evalEnablement disable invalid case based on schema condition', t => {
     ruleValue: 'bar'
   };
   t.is(evalEnablement(uischema, data, undefined, createAjv()), false);
-  t.is(evalEnablement(uischema, { ...data, ruleValue: 'baz' }, undefined, createAjv()), false);
-  t.is(evalEnablement(uischema, { ...data, ruleValue: 'foo' }, undefined, createAjv()), true);
+  t.is(
+    evalEnablement(
+      uischema,
+      { ...data, ruleValue: 'baz' },
+      undefined,
+      createAjv()
+    ),
+    false
+  );
+  t.is(
+    evalEnablement(
+      uischema,
+      { ...data, ruleValue: 'foo' },
+      undefined,
+      createAjv()
+    ),
+    true
+  );
+});
+
+test('isInherentlyEnabled disabled globally', t => {
+  t.false(
+    isInherentlyEnabled(
+      { jsonforms: { readonly: true } },
+      null,
+      null,
+      null,
+      null,
+      null
+    )
+  );
+});
+
+test('isInherentlyEnabled disabled by ownProps', t => {
+  t.false(
+    isInherentlyEnabled(null, { enabled: false }, null, null, null, null)
+  );
+});
+
+test('isInherentlyEnabled enabled by ownProps', t => {
+  t.true(isInherentlyEnabled(null, { enabled: true }, null, null, null, null));
+});
+
+test('isInherentlyEnabled disabled by uischema', t => {
+  t.false(
+    isInherentlyEnabled(
+      null,
+      null,
+      ({ options: { readonly: true } } as unknown) as ControlElement,
+      null,
+      null,
+      null
+    )
+  );
+});
+
+test('isInherentlyEnabled disabled by schema', t => {
+  t.false(
+    isInherentlyEnabled(null, null, null, { readonly: true }, null, null)
+  );
+});
+
+test('isInherentlyEnabled disabled by rule', t => {
+  const leafCondition: LeafCondition = {
+    type: 'LEAF',
+    scope: '#/properties/ruleValue',
+    expectedValue: 'bar'
+  };
+  const uischema: ControlElement = {
+    type: 'Control',
+    scope: '#/properties/value',
+    rule: {
+      effect: RuleEffect.DISABLE,
+      condition: leafCondition
+    }
+  };
```

Comment:
We no longer recommend this old type of rules for a long time. Would be preferable to use a [schema-based rule](https://jsonforms.io/docs/uischema/rules).